pkg/registry/batch/job/strategy.go

```diff
@@ -95,32 +95,13 @@ func (jobStrategy) PrepareForCreate(ctx context.Context, obj runtime.Object) {
 
 	job.Generation = 1
 
-	// While legacy tracking is supported, we use an annotation to mark whether
-	// jobs are tracked with finalizers.
-	addJobTrackingAnnotation(job)
```

Comment:
Let's move this to a different PR (along with validation code).

I think we might actually want to keep adding the annotation for one more release, even if the controller ignores it. The reason is that a cluster might be upgraded to 1.27, then a job is created without the annotation. Then the cluster is rolled back to 1.26. Suddenly, this job is considered not tracked by finalizers, even though it should (it was created in a version when tracking with finalizers was mandatory).

Reply:
So I'll move the strategy and validation job to another PR.  I'll get that done tomorrow AM.

Reply:
These are no longer in the PR.